src/UI/api/utils/sourceStringValidation.ts

```diff
@@ -0,0 +1,21 @@
+export function escapeRegex(inputField: string) {
+  if (typeof inputField === 'boolean') {
+    return inputField;
+  } else {
+    let escapeinputField = inputField.replace(
+      /[-[\]/{}()*+?.,\\^$|#"]/g,
+      '\\$&'
+    );
+    return escapeinputField;
+  }
+}
+
+export function sourceStringValidation(sourceObject: any) {
```

Comment:
Just a single test for this would be nice please - just to ensure everything is being passed through as expected.